tensorflow/core/kernels/data/experimental/random_dataset_op_test.cc

```diff
@@ -0,0 +1,249 @@
+    // the kernel.
+    TF_RETURN_IF_ERROR(
+        CreateDataset(dataset_kernel_.get(), dataset_ctx_.get(), &dataset_));
+
+    // Step 6: Create an iterator in case the test needs to read the output of
+    // the dataset.
+    TF_RETURN_IF_ERROR(
+        CreateIteratorContext(dataset_ctx_.get(), &iterator_ctx_));
+    TF_RETURN_IF_ERROR(dataset_->MakeIterator(iterator_ctx_.get(),
+                                              kIteratorPrefix, &iterator_));
+
+    return Status::OK();
+  }
+
+ protected:
+  // Creates a new `RandomDataset` op kernel.
+  // Doesn't initialize the random seeds because they are inputs, not
+  // attributes.
+  Status CreateRandomDatasetOpKernel(const RandomDatasetParams& dataset_params,
+                                     std::unique_ptr<OpKernel>* op_kernel) {
+    NodeDef node_def = test::function::NDef(
+        kNodeName, name_utils::OpName(RandomDatasetOp::kDatasetType),
+        // Inputs
+        {RandomDatasetOp::kSeed, RandomDatasetOp::kSeed2},
+        // Attributes
+        {{RandomDatasetOp::kOutputTypes, dataset_params.output_dtypes},
+         {RandomDatasetOp::kOutputShapes, dataset_params.output_shapes}});
+    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, op_kernel));
+    return Status::OK();
+  }
+};
+
+RandomDatasetParams FortyTwo() {
+  return {/*seed=*/42,
+          /*seed2=*/42,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+// Change just first seed relative to FortyTwo
+RandomDatasetParams ChangeSeed() {
+  return {/*seed=*/1000,
+          /*seed2=*/42,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+// Change just second seed relative to FortyTwo
+RandomDatasetParams ChangeSeed2() {
+  return {/*seed=*/42,
+          /*seed2=*/1000,
+          /*output_dtypes=*/{DT_INT64},
+          /*output_shapes=*/{PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+class ParameterizedGetNextTest : public RandomDatasetOpTest,
+                                 public ::testing::WithParamInterface<
+                                     GetNextTestCase<RandomDatasetParams>> {};
+
+std::vector<GetNextTestCase<RandomDatasetParams>> GetNextTestCases() {
+  return {{/*dataset_params=*/FortyTwo(),
+           /*expected_outputs=*/GenerateExpectedData(42, 42, kCount)},
+          {/*dataset_params=*/ChangeSeed(),
+           /*expected_outputs=*/GenerateExpectedData(1000, 42, kCount)},
+          {/*dataset_params=*/ChangeSeed2(),
+           /*expected_outputs=*/GenerateExpectedData(42, 1000, kCount)}};
+}
+
+TEST_P(ParameterizedGetNextTest, GetNext) {
+  auto test_case = GetParam();
+  TF_ASSERT_OK(Initialize(&test_case.dataset_params));
+
+  // Can't use DatasetOpsTestBase::CheckIteratorGetNext because the kernel
+  // under test produces unbounded input.
```

Comment:
@rachellim @frreiss As `CheckIteratorGetNext()` does not work here, I think `ITERATOR_GET_NEXT_TEST_P` will not work for this test either.

Reply:
Yes, the current version of `ITERATOR_GET_NEXT_TEST_P` is not appropriate for infinite data sets. I can use the other macros though.